docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,251 @@
+
+```scala
+case opt as Some(foo)
+```
+
+Instead, code that used it should be reverted to use the existing syntax of Scala 2, using `@`:
+
+```scala
+case opt @ Some(foo)
+```
+
+## Switch back to the old context function closure syntax
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+Although opaque types are supposed to provide an opaque abstraction of their underlying types, it is currently too easy to break the abstraction using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+  case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, we introduce a new trait `Matchable` near the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+`Matchable` is currently a marker trait without any methods. Over time, we intend to migrate the methods `getClass` and `isInstanceOf` to it, since these are closely related to pattern-matching.
+
+In the meantime, the compiler will emit warnings when trying to call those methods, or when performing a `match`, on an value that is not a `Matchable`. For example, the above example is now flagged:
+
+```scala
+-- Warning: i7314.scala:6:12 ---------------------------------------------------
+6 |    case a: Array[Int] =>
+  |            ^^^^^^^^^^
+  |            pattern selector should be an instance of Matchable,
+  |            but it has unmatchable type opaques.IArray[Int] instead
+```
+
+Note that the warnings are only active with language mode `3.1-migration` or higher - see the documentation on the [Language Versions](https://dotty.epfl.ch/docs/usage/language-versions.html) to learn how to enable it.
+
+You can read the discussion of this change in the [PR #10670](https://github.com/lampepfl/dotty/pull/10670). You can also read more about it in the [documentation](https://dotty.epfl.ch/docs/reference/other-new-features/matchable.html).
+
+# Tooling improvements
+As we are getting closer to a stable release of Scala 3, the focus increasingly shifts on the tooling available to get started with Scala 3.
+
+For a while now, we are not using the old dottydoc documentation tool for building the documentation. We are developing an entirely new tool, scala3doc, from scratch.
+
+As part of the tooling effort, this new Scala 3 documentation tool is rapidly improved. [PR #10522](https://github.com/lampepfl/dotty/pull/10522) proves that the doctool can generate documentation for the community build projects.
```

Comment:
We can link here to a summaty page of all generated docs from community build: https://scala3doc.virtuslab.com/pr-master-docs/index.html

Reply:
We can also mention that scala3doc is not as slow and fragile as dottydos.